AngularJSAuthentication.Web/app/services/authService.js

```diff
@@ -1,7 +1,8 @@
 ﻿'use strict';
-app.factory('authService', ['$http', '$q', 'localStorageService', 'ngAuthSettings', function ($http, $q, localStorageService, ngAuthSettings) {
+app.factory('authService', ['$q', '$injector', 'localStorageService', 'ngAuthSettings', function ($q, $injector, localStorageService, ngAuthSettings) {
```

Comment:
This is to overcome the issue of circular dependencies. When configuring the interceptor, you make the $httpProvider depend on the interceptor. So if you try to inject $http in the service constructor, you'll get a circular dependency error. To overcome this, I use the $injector service, to postpone injecting the $httpProvider until after everything is ready. 

This is also the reason why $injector.get('authService'); is injected using $injector, since the authService depends on $http as well.